src/views/course-edit-tools/course-feedback.vue

```diff
@@ -0,0 +1,139 @@
+import { storeHandler, locale } from '@/mixins'
+import { jsPDF } from 'jspdf'
+import http from 'axios'
+
+export default {
+  name: 'CourseFeedback',
+
+  mixins: [
+    storeHandler,
+    locale
+  ],
+
+  data () {
+  },
+
+  computed: {
+    ...mapActions([
+      'fetchEnrollment'
+    ]),
+    ...mapGetters([
+      'enrollmentFeedback',
+      'courseId'
+    ])
+  },
+
+  methods: {
+    /**
+     * Function printPDF: prints feedback in a PDF tp download
+     * Author: nv
+     * Last Updated: September 14, 2022
+     */
+    async printPDF () {
+      if (typeof this.enrollmentFeedback !== 'undefined') {
+        /* eslint-disable */
+        const doc = new jsPDF('p','pt')
+        /* eslint-enable */
+        const id = this.courseId
+        let name
+        let feedback = []
+        const x = 30
+        let y = 30
+        const lineheight = doc.getTextDimensions('Sample Text').h + 5
+        let blockheight = 0
+
+        await http.get(`courses/getCourseName?courseId=${id}`)
+          .then(({ data }) => {
+            console.log(data.courseName)
+            name = data.courseName
+          })
+        await http.get(`enrollments/getAllByCourseId?courseId=${id}`)
+          .then(({ data }) => {
+            console.log(data.subs[0].feedback)
+            feedback = data.subs[0].feedback
+          })
+        doc.text(this.y18n('feedback.document.title') + name, x, y)
+        y += lineheight + 20
+        // display of choice answers
+        const choice = JSON.stringify(feedback.map(function (item) {
+          return item['choice']
+        }))
+        doc.text(this.y18n('feedback.document.choices'), x, y)
+        y += lineheight
+        const splitchoice = doc.splitTextToSize(choice, 180)
+        doc.text(splitchoice, 30, y)
+        blockheight = splitchoice.length * lineheight + 5
+        y += blockheight
+        // display of free text answers
+        const freetext = JSON.stringify(feedback.map(function (item) {
+          return item['freetext']
+        }))
+        doc.text(this.y18n('feedback.document.freetext'), x, y)
+        y += lineheight
+        const splitfreetext = doc.splitTextToSize(freetext, 180)
+        doc.text(splitfreetext, 30, y)
+        blockheight = splitfreetext.length * lineheight + 5
+        y += blockheight
+        // display of rating answers
+        const rating = JSON.stringify(feedback.map(function (item) {
+          return item['rating']
+        }))
+        doc.text(this.y18n('feedback.document.rating'), 30, y)
+        y += lineheight
+        const splitrating = doc.splitTextToSize(rating, 180)
+        doc.text(splitrating, 30, y)
+        blockheight = splitchoice.length * lineheight + 5
+        y += blockheight
+
+        const date = doc.getCreationDate()
+        doc.save(date + ' - ' + this.y18n('feedback.document.file') + ' - ' + name + '.pdf')
+      } else {
+        console.log(this.y18n('feedback.document.unavailable'))
+      }
+    }
```

Comment:
the whole function is very hard to read and comprehend, consider splitting it in different small parts